### ISSUE TITLE
job.copy has some slight problems

### ISSUE THREAD

**Issue by rob-c** [ganga-devs/ganga#437]
performing a job.copy on a job which is in a ```completed``` state will potentially give a bad ```outputfiles``` section in the new job.

For a completed job with ```*.txt``` for instance as a namePattern the completed job contains the resolved file from this job and a copy will return a new file which is fully resolved in the ```outputfiles``` which makes sense in terms of copying the actual value of the ```outputfiles``` list.

Should we consider making the ```outputfiles``` empty for jobs which are copying from ```completed``` jobs and inform the user that this is being erased?

This isn't something too serious but it would cause a ```job.copy().submit()``` to fail which is a tad irritating.

**Comment by drmarkwslater** [comment]
I thought (and someone please correct if I'm wrong!) that in the original design for the output files, it stored what it had originally been for just this kind of thing - i.e. if you do a `job.copy()` you get a `new` job with `*.txt` as the output files. I **think** this was implemented with the `subfiles` field.

@alexanderrichards do you remember at all?

**Comment by milliams** [comment]
I think that `j.copy()` should mean "please give me another one like this one" (so I can submit it again, maybe after making a change or two) rather than "give me another object which is equal to this object." This means that indeed we would want to reset `namePattern` to `'*.txt'`.

**Comment by rob-c** [comment]
@drmarkwslater the subfiles field is indeed where the files are stored and I got confused by the summary of the subfiles not reflecting this. It's a feature that the summary descends into the subfiles for the user after a job has completed.
Now, as it stands I'm not sure if this is a bug in the ```DiracFile``` such that the ```setLocation``` is changing the parent job or whether this is a bug in the Job, I suspect the former as the latter is working correctly for ```job.outputfiles``` containing wildcards for ```LocalFile``` objects.

I've had to give this a shake up recently on the Dirac backend so it may have been some funny interaction with this but I also ended up with a strangely malformed job object which wasn't assigning parents so it may have just been a _bad session_ of some form/
